### PR TITLE
Undo faulty serializable resources [GEOS-7513]

### PR DESCRIPTION
**Pull request by NielsCharlier**
Undo faulty serializable resources, Resources.serializable, fix DemoRequestPage and importer module.

For more information see https://osgeo-org.atlassian.net/browse/GEOS-7513

### CODE REVIEW

File: src/community/jdbcstore/src/main/java/org/geoserver/jdbcstore/JDBCResourceStore.java [hunk 1]
```diff
@@ -140,9 +140,7 @@ public String toString() {
      * 
      */    
     protected class JDBCResource implements Resource {
-        
-        private static final long serialVersionUID = 1829574259576354635L;
-        
+
```
Comment (jodygarnett) [inline src/community/jdbcstore/src/main/java/org/geoserver/jdbcstore/JDBCResourceStore.java]: This space is intentionally left blank

File: src/extension/importer/core/src/main/java/org/geoserver/importer/FileData.java [hunk 2]
```diff
@@ -28,7 +29,7 @@
     protected Resource file;
 
     public FileData(Resource file) {
-        this.file = file;
+        this.file = Resources.serializable(file);
```
Comment (jodygarnett) [inline src/extension/importer/core/src/main/java/org/geoserver/importer/FileData.java]: This is not quite what I expected, I expected something like:
````
protected String path;

protected Resource resource(){
     return resources.get( path );
}
````

But I would have to check how FileData is being used, perhaps this is what @cuttlefish was referencing in terms of accidentally loading a geotiff into jdbcstore.

File: src/platform/src/main/java/org/geoserver/platform/resource/Resource.java [hunk 3]
```diff
@@ -20,7 +20,7 @@
  * Resource creation is handled in a lazy fashion, simply use {@link #file()} or {@link #out()} and the resource will be created as required. In a
  * similar fashion setting up a child resource will create any required parent directories.
  */
-public interface Resource extends Serializable {
+public interface Resource {
```
Comment (jodygarnett) [inline src/platform/src/main/java/org/geoserver/platform/resource/Resource.java]: +1 yay!

File: src/extension/importer/core/src/main/java/org/geoserver/importer/FileData.java (hunk 2 above)
Comment (NielsCharlier) [inline src/extension/importer/core/src/main/java/org/geoserver/importer/FileData.java]: As I suggested, I made a serializable resource wrapper for the sake of convenience. This way we hardly need to change any code in the other modules... seemed like a good idea to me, no?

As far as @cuttlefish is concerned, I believe I can help him through a configuration change.

Comment (jodygarnett) [inline src/extension/importer/core/src/main/java/org/geoserver/importer/FileData.java]: Right, got it.

Comment (tbarsballe) [inline src/extension/importer/core/src/main/java/org/geoserver/importer/FileData.java]: It looks like this got missed when the ResourceStore work was initially done, but FileData is a representation of a file that is being loaded into Importer (in other words, not yet added to geoserver), and should actually be using File, not Resource.

**Comment by jodygarnett** [comment]
@tbarsballe I am going to review and merge as is, we can handle the FileData revert in a separate fix.

**Comment by jodygarnett** [comment]
Updated headers and squashed into a single commit on master / 2.9.x.

**Pull request closed without merging**